lib/3scale/backend/stats/delete_job_def.rb

```diff
@@ -0,0 +1,83 @@
+module ThreeScale
+  module Backend
+    module Stats
+      class DeleteJobDef
+        ATTRIBUTES = %i[service_id applications metrics users from to context_info].freeze
+        private_constant :ATTRIBUTES
+        attr_accessor(*ATTRIBUTES)
+
+        def self.attribute_names
+          ATTRIBUTES
+        end
+
+        def initialize(params = {})
+          self.class.attribute_names.each do |key|
+            send("#{key}=", params[key]) unless params[key].nil?
+          end
+        end
+
+        def run_async
+          validate
+          Resque.enqueue(PartitionGeneratorJob, Time.now.getutc.to_f, service_id, applications,
+                         metrics, users, from, to, context_info)
+        end
+
+        def to_json
+          to_hash.to_json
+        end
+
+        def to_hash
+          Hash[self.class.const_get(:ATTRIBUTES).collect { |key| [key, send(key)] }]
+        end
+
+        def validate
+          # from and to valid epoch times
+          unless from.is_a? Integer
+            raise DeleteServiceStatsValidationError.new(service_id, 'from field validation error')
```

Comment:
This pattern is repeated several times.
Would it make sense to extract it to a function, something like this?

```ruby
def raise_error(msg)
  raise DeleteServiceStatsValidationError.new(service_id, msg)
end
```

Reply:
:+1: